src/generate-route-tiles.js

```diff
@@ -0,0 +1,210 @@
+            if (routeTypeLookup.has(routeId)) {
+              routeTypes.add(ROUTE_TYPE_TO_STRING[routeTypeLookup.get(routeId)]);
+            }
+          }
+        }
+      }
+    }
+
+    return routeTypes;
+  };
+
+  for await(const stop of stopsParser) {
+    const stopId = stop['stop_id'];
+    const stopName = stop['stop_name'];
+    const lat = parseFloat(stop['stop_lat']);
+    const lon = parseFloat(stop['stop_lon']);
+
+    if (!isNaN(lat) && !isNaN(lon) && !!stopId) {
+      const properties = {
+        stop_name: stopName,
+        stop_id: stopId,
+      };
+      const routeTypesAtStop = getRouteTypesForStop(stopId);
+      if (routeTypesAtStop.size > 0) {
+        for (const routeType of routeTypesAtStop.values()) {
+          properties[routeType] = true;
+        }
+  
+        const geojson = point([lon, lat], properties);
+  
+        await appendFile(ldGeoJsonPath, JSON.stringify(geojson)+'\n');
+      } else {
+        console.log(`Dropped stop-id: ${stopId}, name: ${stopName} because no routes/trips were detected at the stop`);
+      }
+    }
+  }
+}
+
+async function generateRouteTiles(
+  routelineLookups,
+  unzippedGtfsPath,
+  outputPath,
+) {
+  
+  const routeLinesLDGeoJsonPath = join(outputPath, 'routelines.ldgeojson');
+  if (existsSync(routeLinesLDGeoJsonPath)) {
+    await unlink(routeLinesLDGeoJsonPath);
+  }
+
+  const stopLDGeoJsonPath = join(outputPath, 'stops.ldgeojson');
+  if (existsSync(stopLDGeoJsonPath)) {
+    await unlink(stopLDGeoJsonPath);
+  }
+  
+  console.log('Staring creation of LDGeoJSON');
+  const routesStream = createReadStream(resolve(unzippedGtfsPath, 'routes.txt'), {encoding: 'utf8'});
+  const routesParser = routesStream.pipe(parse({columns: true}));
+  const routeTypes = new Map();
+  for await(const route of routesParser) {
+    await appendRouteLineStringToFile(route, routeLinesLDGeoJsonPath, routelineLookups);
+
+    const routeId = route['route_id'];
+    const routeType = parseInt(route['route_type']);
+    if (!isNaN(routeType)) {
+      routeTypes.set(routeId, routeType);
+    }
+  }
+  console.log('Finished adding route LineStrings LDGeoJSON');
+
+  const stopsStream = createReadStream(resolve(unzippedGtfsPath, 'stops.txt'), {encoding: 'utf8'});
+  const stopsParser = stopsStream.pipe(parse({columns: true}));
+  await appendStops(stopsParser, stopLDGeoJsonPath, routelineLookups, routeTypes);
+
+  console.log('Finished addings stop points to LDGeoJSON');
+
+  if (!shell.which('tippecanoe')) {
+    throw new Error('tippecanoe is not installed and available on PATH');
+  }
+  const routeTilesPath = join(outputPath, 'route-tiles');
+  if (existsSync(routeTilesPath)) {
+    await rm(routeTilesPath, {recursive: true});
+  }
+
+  const stopTilesPath = join(outputPath, 'stop-tiles');
+  if (existsSync(stopTilesPath)) {
+    await rm(stopTilesPath, {recursive: true});
+  }
+
+  const routeLinesCommand = `tippecanoe \
+   -e ${routeTilesPath} \
```

Comment:
This is fragile and will break if the path contains spaces or other characters with special interpretation by the shell. Use the `spawn*` functions from [child_process](https://nodejs.org/dist/latest-v20.x/docs/api/child_process.html) instead of `shell.exec` to avoid this. They take arguments as an array of strings.